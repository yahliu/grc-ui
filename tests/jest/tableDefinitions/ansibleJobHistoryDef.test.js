/* Copyright (c) 2021 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'
import {
  buildAnsibleJobStatus,
  buildViewJobLink
} from '../../../src-web/tableDefinitions/ansibleJobHistoryDef'

describe('tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus', () => {
  it('should return successful anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Successful',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })

  it('should return error anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Error',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })

  it('should return failed anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Failed',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })

  it('should return no anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: '-',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })

  it('should return no anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: '',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })

  it('should return other anisble job status', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Other anisble job status',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildAnsibleJobStatus(item, 'us-en')).toMatchSnapshot()
  })
})

describe('tableDefinitions ansibleJobHistoryDef - buildViewJobLink', () => {
  it('should return valid anisble job link', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Successful',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: 'default/policy-pod-policy-automation-once-hd5xz'
    }
    expect(buildViewJobLink(item, 'us-en')).toMatchSnapshot()
  })

  it('should return empty anisble job link', () => {
    const item = {
      __typename: 'AnsibleAutomationHistory',
      name: 'policy-pod-policy-automation-once-hd5xz',
      message: 'Awaiting next reconciliation',
      status: 'Successful',
      started: '2021-06-03T14:44:43Z',
      finished: null,
      job: ''
    }
    expect(buildViewJobLink(item, 'us-en')).toMatchSnapshot()
  })
})
