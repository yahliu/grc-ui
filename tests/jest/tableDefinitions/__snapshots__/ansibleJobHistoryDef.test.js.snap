// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return error anisble job status 1`] = `
<div>
  <RedExclamationCircleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  Failed
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return failed anisble job status 1`] = `
<div>
  <RedExclamationCircleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  Failed
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return no anisble job status 1`] = `
<div>
  <YellowExclamationTriangleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  No status
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return no anisble job status 2`] = `
<div>
  <YellowExclamationTriangleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  No status
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return other anisble job status 1`] = `
<div>
  <YellowExclamationTriangleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  other anisble job status
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildAnsibleJobStatus should return successful anisble job status 1`] = `
<div>
  <GreenCheckCircleIcon
    tooltip="Awaiting next reconciliation"
  />
   
  Successful
</div>
`;

exports[`tableDefinitions ansibleJobHistoryDef - buildViewJobLink should return empty anisble job link 1`] = `""`;

exports[`tableDefinitions ansibleJobHistoryDef - buildViewJobLink should return valid anisble job link 1`] = `
<a
  href="/search?filters={%22textsearch%22:%22cluster%3Alocal-cluster%20kind%3Ajob%20namespace%3Adefault%20name%3Apolicy-pod-policy-automation-once-hd5xz%22}"
  rel="noopener noreferrer"
  target="_blank"
>
  View job
</a>
`;
